### ISSUE TITLE
dtgePreview is always null?

### ISSUE THREAD

**Issue by jesperordrup** [umco/umbraco-doc-type-grid-editor#123]
Your example

```
@inherits Umbraco.Web.Mvc.UmbracoViewPage
@if (ViewContext.RouteData.Values["dtgePreview"])
{
	// Render preview view
}
else
{
	// Render front end view
}

```

But ViewContext.RouteData.Values["dtgePreview"] is always null

If I render in front its null.
If I open content in backend its null.
If I preview the content using Umbraco Preview (in case I totally misunderstood)

**Comment by leekelleher** [comment]
Thanks for raising this @jesperordrup - odd that it's null. I'll investigate where the bug is ... unless if someone gets there before me? 😃

**Comment by jesperordrup** [comment]
Hello Mr @leekelleher,

Think I found it. You transfer route values correct to actions but not to partials.

Looks great:
`return helper.Action(editorAlias, controllerName, routeValues)`

Looks less great:
`return helper.Partial(content.DocumentTypeAlias, content);`

**Comment by leekelleher** [comment]
@jesperordrup I've been looking at this today. I'm not sure why I put using the `ViewContext.RouteData` check in the docs, it's a bit ugly... I think I was being lazy.

You can use `UmbracoContext.InPreviewMode` in your partial-views, that should work, (we added that in, in v0.6.0).  Although, let me know if you need to know if it's a DTGE preview specifically, (as opposed to Umbraco's full page preview), that would probably require extra code in DTGE to support that.

**Comment by jesperordrup** [comment]
Thanks @leekelleher - perfect.

I was actually confused about what kind of preview it as referring to. If it should be made absolutely clear then it might be better to rename this to:

isBackend
isFrontend
isPreview (being the Umbraco full page preview)

Thanks again
Jesper

**Comment by leekelleher** [comment]
I'm re-opening this one.  While investigating #124, I found that in setting `InPreviewMode = true` causes many other issues. So I'll be looking to find an alternative solution for this. Just a heads up at the moment.

**Comment by leekelleher** [comment]
With DTGE manually setting `InPreviewMode` causes more headaches than it's worth, so I've (hopefully) fixed it in PR #127.  Ultimately it'll be a check for `ViewData["dtgePreview"]` in your partial-views. This should be consistent across both the `helper.Action` and `helper.Partial` calls.

As for other useful preview flags that can be checked, Umbraco already has: `UmbracoContext.InPreviewMode` (for the full page preview) and `UmbracoContext.IsFrontEndUmbracoRequest`.  Hope this helps?

Closing off this ticket, any discussion about the implementation can be had on PR #127.
